src/main/java/org/folio/fqm/repository/EntityTypeRepository.java

```diff
@@ -52,12 +56,20 @@ public Optional<EntityType> getEntityTypeDefinition(UUID entityTypeId) {
 
     Field<String> definitionField = field("definition", String.class);
 
-    return jooqContext
+    Optional<EntityType> entityTypeOptional = jooqContext
       .select(definitionField)
       .from(table(TABLE_NAME))
       .where(field(ID_FIELD_NAME).eq(entityTypeId))
       .fetchOptional(definitionField)
       .map(this::unmarshallEntityType);
+
+    EntityType entityType = entityTypeOptional.orElseThrow(() -> new EntityTypeNotFoundException(entityTypeId));
+
+    String customFieldsEntityTypeId = entityType.getCustomFieldEntityTypeId();
+    if (customFieldsEntityTypeId != null) {
+      entityType.getColumns().addAll(fetchColumnNamesForCustomFields(UUID.fromString(customFieldsEntityTypeId)));
+    }
+    return Optional.of(entityType);
```

Comment:
Not sure we want to handle the exception throwing here (and if we do, there's no reason to return an `Optional`).  I'd let the consuming method handle the `Optional` and leave the not found exception for the consuming method, as it was before.  An easy way to do this is with a `map`:

```suggestion
    return jooqContext
      .select(definitionField)
      .from(table(TABLE_NAME))
      .where(field(ID_FIELD_NAME).eq(entityTypeId))
      .fetchOptional(definitionField)
      .map(this::unmarshallEntityType)
      .map(entityType -> {
        String customFieldsEntityTypeId = entityType.getCustomFieldEntityTypeId();
        if (customFieldsEntityTypeId != null) {
          entityType.getColumns().addAll(fetchColumnNamesForCustomFields(UUID.fromString(customFieldsEntityTypeId)));
        }
        return entityType;
      });
```